util/chplenv/chpl_llvm.py

```diff
@@ -163,11 +163,14 @@ def get_llvm_config():
     return llvm_config
 
 @memoize
-def validate_llvm_config():
+def validate_llvm_config(llvm_config=None):
     llvm_val = get()
-    llvm_config = get_llvm_config()
+    # We pass in llvm_config if has already been computed (so we don't
+    # end up in an infinite loop).
+    if llvm_config is None:
+      llvm_config = get_llvm_config()
     if llvm_val == 'system':
-        if llvm_config == 'none':
+        if not llvm_config or llvm_config == 'none':
```

Comment:
I think this line would be more clear as `if llvm_config == '' or llvm_config == 'none':`.

Reply:
Sure, updated in 4ca1e4dc8441d3bbc87ffde412129a4bccfaf0a0